frontend/public/extend/devconsole/components/codebases/codebaselist.jsx

```diff
@@ -0,0 +1,146 @@
+import * as _ from 'lodash-es';
+import * as React from 'react';
+
+import { ColHead, DetailsPage, List, ListHeader, ListPage, ResourceRow } from '../../../../components/factory';
+import { Kebab, navFactory, LabelList, SectionHeading, ResourceIcon, ResourceLink, ResourceSummary, Selector } from '../../../../components/utils';
+
+const menuActions = [Kebab.factory.ModifyPodSelector, ...Kebab.factory.common];
+
+const ServiceIP = ({s}) => {
+  const children = _.map(s.spec.ports, (portObj, i) => {
+    const clusterIP = s.spec.clusterIP === 'None' ? 'None' : `${s.spec.clusterIP}:${portObj.port}`;
+    return <div key={i}>{clusterIP}</div>;
+  });
+
+  return children;
+};
+
+const ComponentHeader = props => <ListHeader>
```

Comment:
Why is the component named `ComponentHeader`? Shouldn't it be `CodebasesHeader` or `CodebasesListHeader`?

Reply:
Also use parentheses to wrap multiline JSX return statements. 
```suggestion
const ComponentHeader = (props) => (
  <ListHeader>
  </ListHeader>
)
```

Reply:
Add prop types using typescript interfaces. Also add proper type definitions for your components.
```suggestion
const ComponentHeader: React.FunctionComponent = (props: HeaderProps) => (
  <ListHeader>
)
```

Reply:
Add prop types using typescript interfaces. Also add proper type definitions for your components.
```suggestion
const ComponentHeader: React.FunctionComponent<HeaderProps> = (props: HeaderProps) => (
  <ListHeader>
)
```

Reply:
Run prettier!